dashboard/app/controllers/script_levels_controller.rb

```diff
@@ -199,9 +199,31 @@ def load_section
     end
   end
 
+  def select_level
+    # If there's only one level in this scriptlevel, use that
+    return @script_level.levels[0] if @script_level.levels.length == 1
+
+    # If they've tried at least one variant before, use the most recently attempted
+    # (unless overridden by a force_reload query string param)
+    if current_user && !params[:force_reload]
+      last_attempt = current_user.last_attempt_for_any(@script_level.levels)
+      return last_attempt.level if last_attempt
+    end
+
+    # Otherwise return the active level with the lowest id. Active defaults to true
+    return @script_level.levels.min_by(&:id) unless @script_level.properties
+
+    properties = JSON.parse(@script_level.properties)
+    @script_level.levels.sort_by(&:id).each do |level|
+      return level unless properties[level.name] && properties[level.name]['active'] == false
```

Comment:
It's also not clear to me that `id` is the best attribute to sort by; the incremental nature of ids is a side effect of one particular method of enforcing uniqueness, not a feature in and of itself. Creation time might be what we really want here.